src/linking/remove_internal_symbols.cpp

```diff
@@ -19,45 +19,44 @@ Author: Daniel Kroening
 
 #include "static_lifetime_init.h"
 
-void get_symbols_rec(
+static void get_symbols(
   const namespacet &ns,
   const symbolt &symbol,
   find_symbols_sett &dest)
 {
-  dest.insert(symbol.name);
+  std::vector<const symbolt *> working_set;
```

Comment:
It would have used a `forward_list`, but it's probably ok.

Reply:
I like the density of the vector, and the fact that it won't actually do memory allocation once the working set has reached some size.

Reply:
Quick experiment: 14.7s vs. 16.2s, for total goto-cc time on a larger .c file, in favour of the vector.